setup.cfg

```diff
@@ -0,0 +1,128 @@
+license = Apache License 2.0
+long_description = file: README.rst
+long_description_content_type = text/x-rst
+classifiers =
+    License :: OSI Approved :: Apache Software License
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+
+[options]
+python_requires = >=3.8
+packages = find:
+# =src is interpreted as {"": "src"}
+# as per recommendation here https://hynek.me/articles/testing-packaging/
+package_dir =
+    =src
+    
+setup_requires =
+    setuptools_scm[toml]>=6.2
+
+# Specify any package dependencies below.
+# install_requires =
+#     numpy
+#     scipy
+
+[options.extras_require]
+# For development tests/docs
+dev =
+    black==22.6.0
+    pipdeptree
+    flake8-isort
+    isort>5.0
+    mypy
+    pre-commit
+    pytest-cov
+    sphinx-rtd-theme-github-versions
+    tox
+    setuptools_scm[toml]>=6.2
+    types-mock
+    sphinx-rtd-theme-github-versions
+    sphinx-apischema
+
+[options.packages.find]
+where = src
+# Don't include our tests directory in the distribution
+exclude = tests
+
+# Specify any package data to be included in the wheel below.
+# [options.package_data]
+# python3_pip_skeleton =
+#     subpackage/*.yaml
+
+[options.entry_points]
+# Include a command line script
+console_scripts =
+    python3-pip-skeleton = python3_pip_skeleton.__main__:main
+
+[mypy]
+# Ignore missing stubs for modules we use
+ignore_missing_imports = True
+
+[isort]
+profile=black
+float_to_top=true
+
+[flake8]
+# Make flake8 respect black's line length (default 88),
+max-line-length = 88
+extend-ignore =
+    E203,  # See https://github.com/PyCQA/pycodestyle/issues/373
+    F811,  # support typing.overload decorator
+    F722,  # allow Annotated[typ, some_func("some string")]
+exclude =
+    ui_*
+    .tox
+    .venv
+
+[tool:pytest]
+# Run pytest with all our checkers, and don't spam us with massive tracebacks on error
+addopts =
+    --tb=native -vv --doctest-modules --doctest-glob="*.rst"
+    --cov=python3_pip_skeleton --cov-report term --cov-report xml:cov.xml
+# https://iscinumpy.gitlab.io/post/bound-version-constraints/#watch-for-warnings
+filterwarnings = error
+# Doctest python code in docs, python code in src docstrings, test functions in tests
+testpaths =
+    docs src tests
+
+[coverage:run]
+data_file = /tmp/python3_pip_skeleton.coverage
+
+[coverage:paths]
+# Tests are run from installed location, map back to the src directory
+source =
+    src
+    **/site-packages/
+
+# Use tox to provide parallel linting and testing
+# NOTE that we pre-install all tools in the dev dependencies (including tox).
+# Hence the use of allowlist_externals instead of using the tox virtualenvs.
+# This ensures a match between developer time tools in the IDE and tox tools.
+[tox:tox]
+minversion = 3.7
+skipsdist=true
+skipinstall=true
+
+[testenv:{pre-commit,mypy,pytest,docs}]
+passenv =
+    PYTHONPATH
+    HOME
+setenv =
+    QT_QPA_PLATFORM = offscreen
```

Comment:
What is this for?

Reply:
I think the setenv bit can go as there is no QT in here

Reply:
agreed, this is a holdover from dls-pmac-control where I originally tested.